File: src/java/com/kitware/pulse/cdm/patient/actions/SERespiratoryMechanicsModification.java
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

package com.kitware.pulse.cdm.patient.actions;

import com.kitware.pulse.cdm.bind.PatientActions.RespiratoryMechanicsModificationData;
import com.kitware.pulse.cdm.system.physiology.SERespiratoryMechanicsModifiers;

public class SERespiratoryMechanicsModification extends SEPatientAction
  {
    protected SERespiratoryMechanicsModifiers modifiers = null;
    protected String modifiersFile = "";
    protected boolean incremental = false;

    public SERespiratoryMechanicsModification()
    {
      super();
    }

    public SERespiratoryMechanicsModification(SERespiratoryMechanicsModification other)
    {
      copy(other);
    }

    public void copy(SERespiratoryMechanicsModification other)
    {
      super.copy(other);
      if (other.modifiers != null)
        this.getModifiers().copy(other.modifiers);
      this.modifiersFile = other.modifiersFile;
      this.incremental = other.incremental;
    }

    public void clear()
    {
      super.clear();
      if (this.modifiers != null)
        this.modifiers.clear();
      this.modifiersFile = "";
      this.incremental = false;
    }
    
    public static void load(RespiratoryMechanicsModificationData src, SERespiratoryMechanicsModification dst)
    {
      if (src.hasModifiers())
        SERespiratoryMechanicsModifiers.load(src.getModifiers(), dst.getModifiers());
      else if (src.getModifiersFile().isEmpty())
        dst.setModifiersFile(src.getModifiersFile());
      dst.incremental = src.getIncremental();
    }
    public static RespiratoryMechanicsModificationData unload(SERespiratoryMechanicsModification src)
    {
      RespiratoryMechanicsModificationData.Builder dst = RespiratoryMechanicsModificationData.newBuilder();
      unload(src,dst);
      return dst.build();
    }
    protected static void unload(SERespiratoryMechanicsModification src, RespiratoryMechanicsModificationData.Builder dst)
    {
      if(src.hasModifiers())
        dst.setModifiers(SERespiratoryMechanicsModifiers.unload(src.getModifiers()));
      else if (src.hasModifiersFile())
        dst.setModifiersFile(src.getModifiersFile());
      dst.setIncremental(src.incremental);
    }

    public boolean isValid()
    {
      return hasModifiers() || hasModifiersFile();
    }

    public boolean getIncremental() { return this.incremental; }
    void setIncremental(boolean b) { this.incremental = b; }

    public boolean hasModifiers()
    {
      return this.modifiers != null;
    }
    public SERespiratoryMechanicsModifiers getModifiers()
    {
      if (this.modifiers == null)
        this.modifiers = new SERespiratoryMechanicsModifiers();
      return this.modifiers;
    }

    public boolean hasModifiersFile()
    {
      return this.modifiersFile != null && !this.modifiersFile.isEmpty();
    }
    public String getModifiersFile()
    {
      return this.modifiersFile;
    }
    public void setModifiersFile(String s)
    {
      this.modifiersFile = s;
    }

    public String toString()
    {
      String str = "Respiratory Mechanics Modification";
      if (this.hasModifiersFile())
        str += "\n\tModifiers File: " + this.modifiersFile;
      else if (hasModifiers())
      {
        str += modifiers.toString();
      }
      str += "\n\tIncremental: " + this.incremental;
      return str;
    }
  }

File: src/csharp/pulse/cdm/patient/actions/SERespiratoryMechanicsModification.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

namespace Pulse.CDM
{
  public class SERespiratoryMechanicsModification : SEPatientAction
  {
    protected SERespiratoryMechanicsModifiers modifiers = null;
    protected string modifiersFile = "";
    protected bool incremental = false;

    public SERespiratoryMechanicsModification()
    {

    }

    public SERespiratoryMechanicsModification(SERespiratoryMechanicsModification other)
    {
      Copy(other);
    }

    public void Copy(SERespiratoryMechanicsModification other)
    {
      base.Copy(other);
      if (other.modifiers != null)
        this.GetModifiers().Copy(other.modifiers);
      this.modifiersFile = other.modifiersFile;
      this.incremental = other.incremental;
    }

    public bool GetIncremental() { return incremental; }
    public void SetIncremental(bool b) { incremental = b; }

    public override void Clear()
    {
      base.Clear();
      if (this.modifiers != null)
        this.modifiers.Clear();
      this.modifiersFile = "";
      incremental = false;
    }

    public override bool IsValid()
    {
      return HasModifiers() || HasModifiersFile();
    }

    public bool HasModifiers()
    {
      return this.modifiers != null;
    }
    public SERespiratoryMechanicsModifiers GetModifiers()
    {
      if (this.modifiers == null)
        this.modifiers = new SERespiratoryMechanicsModifiers();
      return this.modifiers;
    }

    public bool HasModifiersFile()
    {
      return !string.IsNullOrEmpty(this.modifiersFile);
    }
    public string GetModifiersFile()
    {
      return this.modifiersFile;
    }
    public void SetModificationFile(string s)
    {
      this.modifiersFile = s;
    }

    public override string ToString()
    {
      string str = "Respiratory Mechanics Modification";
      if (this.HasModifiersFile())
        str += "\n\tModifiers File: " + this.modifiersFile;
      else if (HasModifiers())
      {
        str += modifiers.ToString();
      }
      str += "\n\tIncremental: " + this.incremental;
      return str;
    }
  }
}
